File: quarkus-app/src/main/java/ec/com/eurofish/model/HaccpModel.java
package ec.com.eurofish.model;

import io.vertx.core.json.JsonObject;
import io.vertx.mutiny.sqlclient.Row;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class HaccpModel {
    String lot;
    // Integer state;
    JsonObject body;

    public static HaccpModel from(Row row) {
        return HaccpModel.builder()
                .lot(row.getString("lot"))
                .body(row.getJsonObject("body"))
                .build();
    }
}

File: node-app/src/vessels/domain/model/haccp-model.ts

export class HaccpModel {
    lot: string;
    body: Record<string, any>;
    
    constructor(lot: string, body: any) {
        this.lot = lot;
        this.body = body;
    }

    static fromRow(row: any): HaccpModel {
        return new HaccpModel(row.lot, row.body);
    }
}